packages/react-i18n/src/index.tsx

```diff
@@ -61,22 +125,92 @@ export const withI18n = createHigherOrderComponent< I18nReact >( ( InnerComponen
 	};
 }, 'withI18n' );
 
+interface MakeContextValueOptions {
+	filters: I18nFilters;
+}
+
+/**
+ * Bind an I18n function to its instance
+ *
+ * @param i18n I18n instance
+ * @param fnName '__' | '_n' | '_nx' | '_x'
+ * @param options Make context value options object
+ * @returns Bound I18n function with applied transformation hooks
+ */
+function bindI18nFunction(
+	i18n: I18n,
+	fnName: '__' | '_n' | '_nx' | '_x',
+	options: MakeContextValueOptions
+) {
+	const boundFn = i18n[ fnName ].bind( i18n );
+
+	if ( ! hasI18nFilters() ) {
+		return boundFn;
+	}
+
+	return ( ...args: ( string | number )[] ) => {
+		const filteredArguments = options.filters.applyFilters(
+			'preTranslation',
+			args,
+			fnName,
+			options
+		);
+
+		return options.filters.applyFilters(
+			'postTranslation',
+			boundFn( ...filteredArguments ),
```

Comment:
Little optimization opportunity: now when you're creating a new wrapped function, you can call `i18n[ fnName ]` directly and `boundFn` doesn't need to be created at all.

Reply:
I preserved the binding from the original code, but looking at `@wordpress/i18n` seems like it's not needed at all. Removed it in https://github.com/Automattic/wp-calypso/pull/44743/commits/693f3c77549aa3b01c1937d54393582be75a9ee8.